lib/lambda_ethereum_consensus/validator/validator.ex

```diff
@@ -223,42 +200,144 @@ defmodule LambdaEthereumConsensus.Validator do
           :ok
   def sync_committee_message_broadcast(
         %{index: validator_index, keystore: keystore},
-        current_duty,
+        %{subnet_ids: subnet_ids} = current_duty,
         slot,
         head_root
       ) do
-    %{subnet_ids: subnet_ids} = current_duty
-
-    head_state = BlockStates.get_state_info!(head_root).beacon_state |> go_to_slot(slot)
     log_debug(validator_index, "broadcasting sync committee message", slot: slot)
 
-    head_state
-    |> get_sync_committee_message(head_root, validator_index, keystore.privkey)
+    message =
+      get_sync_committee_message(current_duty, slot, head_root, validator_index, keystore.privkey)
+
+    message
     |> Gossip.SyncCommittee.publish(subnet_ids)
     |> log_info_result(validator_index, "published sync committee message", slot: slot)
+
+    aggregation = current_duty |> Map.get(:aggregation, [])
+
+    if Enum.any?(aggregation, &(not &1.aggregated?)) do
+      log_debug(validator_index, "collecting for future contribution", slot: slot)
+
+      aggregation
+      |> Enum.map(& &1.subcommittee_index)
+      |> Gossip.SyncCommittee.collect(message)
+      |> log_debug_result(validator_index, "collected sync committee messages", slot: slot)
+    end
   end
 
   @spec get_sync_committee_message(
-          Types.BeaconState.t(),
+          Duties.sync_committee_duty(),
+          Types.slot(),
           Types.root(),
           Types.validator_index(),
           Bls.privkey()
         ) ::
           Types.SyncCommitteeMessage.t()
-  def get_sync_committee_message(head_state, head_root, validator_index, privkey) do
-    epoch = Accessors.get_current_epoch(head_state)
-    domain = Accessors.get_domain(head_state, Constants.domain_sync_committee(), epoch)
+  def get_sync_committee_message(
+        %{validator_index: validator_index, message_domain: domain},
+        slot,
+        head_root,
+        validator_index,
+        privkey
+      ) do
     signing_root = Misc.compute_signing_root(head_root, domain)
     {:ok, signature} = Bls.sign(privkey, signing_root)
 
     %Types.SyncCommitteeMessage{
-      slot: head_state.slot,
+      slot: slot,
       beacon_block_root: head_root,
       validator_index: validator_index,
       signature: signature
     }
   end
 
+  @spec publish_sync_aggregate(
+          t(),
+          Duties.sync_committee_duty(),
+          %{},
+          Types.slot()
+        ) :: :ok
+  def publish_sync_aggregate(
+        %{index: validator_index, keystore: keystore},
+        %{validator_index: validator_index} = duty,
+        sync_subcommittee_participants,
+        slot
+      ) do
+    for %{subcommittee_index: subnet_id} = aggregation_duty <- duty.aggregation do
+      case Gossip.SyncCommittee.stop_collecting(subnet_id) do
```

Comment:
Yes, this is exactly what happens, we discussed this briefly on Friday's sync if I remember correctly, this happens also with attestations and that's why i didn't wanted to tackle it yet and just leave it as it was, here is an issue to handle it correctly: #1286 (BTW this is a p2 because outside of kurtosis we have very low probabilities to have more than one of our aggregators in a particular slot)

Reply:
Added the TODO comment and the task in it.